pkg/imagehandler/imagehandler.go

```diff
@@ -80,34 +81,30 @@ func (f *imageFileSystem) getBaseImage(initramfs bool) baseFile {
 	}
 }
 
-func (f *imageFileSystem) getNameForKey(key string) (name string, err error) {
-	if img, exists := f.images[key]; exists {
-		return img.name, nil
-	}
-	rand, err := uuid.NewRandom()
-	if err == nil {
-		name = rand.String()
-	}
-	return
-}
-
 func (f *imageFileSystem) ServeImage(key string, ignitionContent []byte, initramfs, static bool) (string, error) {
-	size, err := f.getBaseImage(initramfs).Size()
+	baseImage := f.getBaseImage(initramfs)
+	size, err := baseImage.Size()
 	if err != nil {
 		return "", InvalidBaseImageError{cause: err}
 	}
 
-	f.mu.Lock()
-	defer f.mu.Unlock()
-
-	name := key
+	var name string
 	if !static {
-		name, err = f.getNameForKey(key)
+		basePath, err := baseImage.CheckSum()
 		if err != nil {
-			return "", err
+			return "", InvalidBaseImageError{cause: err}
 		}
+
+		ignitionCheckSum := sha256.Sum256(ignitionContent)
+		name = fmt.Sprintf("/%s/%s", basePath, hex.EncodeToString(ignitionCheckSum[:]))
```

Comment:
No danger of this URL getting too long for stuff to handle?
I guess we are currently caching everything in ironic and not passing this URL to BMCs.

Reply:
Yep, we do not pass it to BMCs. Otherwise, the URL length should not exceed 2*64 + length of the host, so not even 256.